apps/ergw_core/src/gtp_context.erl

```diff
@@ -418,9 +418,9 @@ handle_event({call, From},
 handle_event({call, From},
 	     {sx, #pfcp{type = session_report_request,
 			ie = #{report_type := #report_type{upir = 1}}}},
-	     State, #{pfcp := PCtx} = Data0) ->
-    Data = close_context(both, up_inactivity_timeout, State, Data0),
-    {next_state, shutdown, Data, [{reply, From, {ok, PCtx}}]};
+	     State, #{pfcp := PCtx} = Data) ->
+    gen_statem:reply(From, {ok, PCtx}),
```

Comment:
Is there a reason not to return a {reply, ...} value instead of manually replying? Is delete_context already returning something?

If this is the case, I'd prefer result values to calls/casts to be explicitly stated in the body of the appropriate handle_X function. It becomes difficult to follow the messages otherwise.

Reply:
1. gen_statem does not have a `{reply, ...}` return value, it would need a reply action instead
2. `delete_context/4` is returning `{next_state,...}` already. Passing `{ok, PCtx}` into that would need changes to delete_context and I want to keep the changes as local and minimal as possible.

Reply:
1. Thanks, I was forgetting that it's included in the list of actions.
2. Yes, that is the behaviour I don't like :) I'd like to see a explicit action description in the body of the handle_ function. But I agree that this may not be the best case to start working on this.